lib/xcodeproj/project/object/helpers/file_references_factory.rb

```diff
@@ -180,15 +180,16 @@ def new_subproject(group, path, source_tree)
             ref = new_file_reference(group, path, source_tree)
             ref.include_in_index = nil
 
-            product_group_ref = find_products_group_ref(group, true)
+            product_group_ref = group.project.new(PBXGroup)
```

Comment:
does Xcode keep adding a "products" ref for each subproject?

Reply:
Yes, Xcode does add a "Products" group for every subproject.